### PR TITLE
Add companies house oauth2 views

### PR DESCRIPTION
**Pull request by richtier**
[this task](https://uktrade.atlassian.net/browse/ED-2026)
[this task](https://uktrade.atlassian.net/browse/ED-2030)

### CODE REVIEW

File: company/templates/companies-house-oauth2-callback.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{% extends "govuk_layout.html" %}
```
Comment (richtier) [inline company/templates/companies-house-oauth2-callback.html]: placeholder. will replace with nice UI once it is designed.

**Comment by richtier** [comment]
code climate must be drunk:

image

File: .codeclimate.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    enabled: true
+  duplication:
+    enabled: true
+    config:
+      languages:
+      - python:
+ratings:
+  paths:
+    - "**.py"
+exclude_paths:
```
Comment (konradko) [inline .codeclimate.yml]: looks like these exclusions are not working: https://codeclimate.com/github/uktrade/directory-ui-buyer/pull/183

File: .codeclimate.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+- "**/tests/"
+- "migrations/"
+- "Dockerfile"
+- "docker/"
+- "README.md"
+- "LICENCE"
+- "*.yml"
+- "**/conftest.py"
+- "**/*.html"
+- "**/static/"
```
Comment (konradko) [inline .codeclimate.yml]: we only want to exclude 3rd party static, so `static/vendor`

File: .codeclimate.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  duplication:
+    enabled: true
+    config:
+      languages:
+      - python:
+ratings:
+  paths:
+    - "**.py"
+exclude_paths:
+- "**/tests/"
```
Comment (konradko) [inline .codeclimate.yml]: looking at https://codeclimate.com/github/uktrade/directory-ui-buyer/issues it seems like admin/tests/test_admin.py is still included

Comment (richtier) [inline .codeclimate.yml]: I think that's because /issues is showing issues in master, not in my branch

Comment (konradko) [inline .codeclimate.yml]: yeah let's see what happens after merge

File: .codeclimate.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+- "Dockerfile"
+- "docker/"
+- "README.md"
+- "LICENCE"
+- "*.yml"
+- "**/conftest.py"
+- "**/*.html"
```
Comment (konradko) [inline .codeclimate.yml]: why we don't want to use codeclimate on templates and css we develop?

File: company/views.py [hunk 6]
```diff
@@ -451,3 +419,52 @@ def dispatch(self, request, *args, **kwargs):
...
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['redirect_uri'] = self.redirect_uri
+        return kwargs
+
+    def form_valid(self, form):
+        response = api_client.company.verify_with_companies_house(
+            sso_session_id=self.request.sso_user.session_id,
+            access_token=form.oauth2_response.json()['access-token']
```
Comment (konradko) [inline company/views.py]: isn't this `authorization_code`?

File: requirements.ini [hunk 7]
```diff
@@ -10,7 +10,7 @@ phonenumberslite==7.7.4
 pillow==4.0.0
 django-cors-headers==2.0.2
 -e git+https://github.com/uktrade/directory-validators.git@v2.3.0#egg=directory_validators
--e git+https://github.com/uktrade/directory-api-client.git@v6.0.0#egg=directory_api_client
+-e git+https://github.com/uktrade/directory-api-client.git@ED-2035-verify-companies-house-code#egg=directory_api_client
```
Comment (richtier) [inline requirements.ini]: to update to `v7.1.0` once the api-client PR is merged.

File: .codeclimate.yml (hunk 5 above)
Comment (richtier) [inline .codeclimate.yml]: `css` is compiled from `.scss` files. codeclimate will be running against the scss files.

can codeclimate analyse HTML? it seemed to be giving grey boxes for HTML file ratings. I removed HTML files to remove some noise,

File: company/views.py (hunk 6 above)
Comment (richtier) [inline company/views.py]: as discussed, nope - `oauth2_response` is this one:

image

ah I see now its `access_token`, not `access-token`. Will chane.

File: .codeclimate.yml (hunk 5 above)
Comment (konradko) [inline .codeclimate.yml]: 👍

**Pull request merged**